Patch 3

naomi-lgbt opened the pull request:
Adding help command and updating bootstrap file to run it.

naomi-lgbt committed:
Create help.js

New help function

naomi-lgbt committed:
Update bootstrap.js

Add help requirement

bradtaniguchi commented on src/commands/bootstrap.js:
```diff
@@ -1,5 +1,5 @@
 const eightball = require('./eightball');
-
+const help = require('./help')
```
This is imported, but not added to the list of commands, so this command wont be available.

naomi-lgbt replied:
Oops. Missed that line.

naomi-lgbt replied:
Fixed

bradtaniguchi commented on src/commands/help.js:
```diff
@@ -0,0 +1,35 @@
+
+  command: function help(message) {
+    console.log('help', message);
+      const helpEmbed = new Discord.MessageEmbed()
```
Wrap a `try{} catch(){}` around the sending of the message so the bot doesn't crash.
in the `catch` be sure to log the error.

naomi-lgbt replied:
Added a catch feature

bradtaniguchi commented on src/commands/help.js:
```diff
@@ -0,0 +1,35 @@
+      .addFields(
+        { name: '\u200B', value: '\u200B' },
+        {name: "Commands", value: "Here are my currently available commands!"},
+        {name: "!eightball <optional string>", value: "Returns a response from a magic 8 ball toy"},
```
The magic 8 ball is not a toy! Its a magic pool ball!

naomi-lgbt replied:
FIXED

bradtaniguchi commented on src/commands/help.js:
```diff
@@ -0,0 +1,35 @@
+   *{name: "Command Name", value: "Command Description"},
+   */
+
+  command: function help(message) {
```
Formatting for this code doesn't look right, as the indentation is off. Are you running this code in an editor that can format with prettier on save? Once automated linting is setup this will error out.

naomi-lgbt replied:
Ran through prettier

naomi-lgbt committed:
Update help.js

Added error catch on .send(helpEmbed)
Switched .send(helpEmbed) to direct message, write to channel that message was sent.
Formatted using prettier
Properly defined what a magic 8 ball is 😎

naomi-lgbt committed:
Update bootstrap.js

add help to commands object

bradtaniguchi commented on src/commands/help.js:
```diff
@@ -0,0 +1,46 @@
+      .send(helpEmbed)
+      .catch((err) => message.channel.send(err.message));
+    message.channel.send(message.author.tag + ", I've sent you a message!");
+  },
```
Instead of using `.catch` we should use `async/await` in a `try {} catch() {}`. I'm on the fence about sending a message with the error, id rather log it and just send nothing back to the user.

naomi-lgbt replied:
Okay, I believe I implemented this change correctly?

bradtaniguchi commented on src/commands/help.js:
```diff
@@ -0,0 +1,46 @@
+  /**
+   *Use this format to add new commands:
+   *{name: "Command Name", value: "Command Description"},
+   */
```
Combine these two comments into 1

naomi-lgbt replied:
Done :)

naomi-lgbt committed:
Combine Comments

naomi-lgbt committed:
Merge branch 'master' into patch-3

naomi-lgbt committed:
Add async/await

Did I do this right?

bradtaniguchi reviewed (commented):
Make 1 change, then LGTM!

bradtaniguchi commented on src/commands/help.js:
```diff
@@ -0,0 +1,44 @@
+
+      message.author.send(helpEmbed);
+    } catch (error) {
+      console.log(error);
```
Change to `console.error` instead of `console.log`

naomi-lgbt replied:
Done!
Heh - I've never actually heard of console.log alternatives until today 😬

Ryuno-Ki replied:
https://developer.mozilla.org/en-US/docs/Web/API/Console#Methods

naomi-lgbt committed:
console.error
